### PR TITLE
Register OSG APs (SOFTWARE-4791)

### PR DESCRIPTION
**Pull request by brianhlin**
Hosts/DNs/contacts mostly grabbed from https://github.com/opensciencegrid/osg-flock/blob/master/flock.opensciencegrid.org/flocking-hosts.yml#L7-L17

@rynge I threw the few that had dev or test in their names into ITB groups

### CODE REVIEW

**Comment by matyasselmeci** [comment]
This is a lot for one person to review. @edquist can we split this up? You take "topology/American Museum of Natural History through "topology/University of California San Diego" and I take "topology/University of Chicago" through "topology/University of Wisconsin Milwaukee" ?

**Comment by edquist** [comment]
I did look (gloss) this over, but @brianhlin, how would you like us to review this?  Do you want us to verify everything added here against the flocking-hosts.yml you mentioned from the osg-flock repo?

**Comment by brianhlin** [comment]
>  Do you want us to verify everything added here against the flocking-hosts.yml you mentioned from the osg-flock repo?

Yeah, that seems like the most reasonable way to do it. I had to make a bunch of educated guesses on new sites/resource groups so if anything looks offensive there, that'd be worth bringing up, too.

File: topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml [hunk 1]
```diff
@@ -13,63 +13,330 @@ Resources:
...
           ID: 0a22bab3de2d83d723811e3fb1ebca904e924a97
+      Security Contact:
+        Primary:
+          Name: Judith Stephen
+          ID: a418fbc5dd33637bba264c01d84d52dd317f2813
         Secondary:
+          Name: Lincoln Bryant
+          ID: 0a22bab3de2d83d723811e3fb1ebca904e924a97
+    FQDN: login04.osgconnect.net
+    DN: /DC=org/DC=incommon/C=US/ST=IL/L=Chicago/O=University of Chicago/OU=IT Services - Self Enrollment/CN=condor-flocking-cert.grid.uchicago.edu
```
Comment (matyasselmeci) [inline topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml:29]: Is this the right DN? Shouldn't the CN match the hostname?

Comment (brianhlin) [reply topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml:29]: A lot of the flocking hosts appear to share a cert. I imagine that it's got a lot of SANs but @rynge or @LincolnBryant would know how this actually works.

File: topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml [hunk 2]
```diff
@@ -13,63 +13,330 @@ Resources:
     ContactLists:
       Administrative Contact:
         Primary:
+          Name: Judith Stephen
```
Comment (matyasselmeci) [inline topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml:16]: Can you also make a PR for flocking-hosts.yml in osg-flock to make Judith the admin for the hosts she's going to be the primary contact for?

File: topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml [hunk 3]
```diff
@@ -13,63 +13,330 @@ Resources:
...
 
+  UChicago_OSGConnect_cms:
+    Active: true
+    Description: The CMS CI Connect access point
+    ID: 1200
+    ContactLists:
+      Administrative Contact:
+        Primary:
+          ID: 82122a855bc0fe808efd91e402e7975a39511d7b
+          Name: Kenyi Hurtado Anampa
```
Comment (matyasselmeci) [inline topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml:224]: Same change for flocking-hosts.yml except using Kenyi

File: topology/University of Chicago/UChicago/UChicago_OSGConnect_ITB.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          Name: Judith Stephen
+          ID: a418fbc5dd33637bba264c01d84d52dd317f2813
+        Secondary:
+          Name: Lincoln Bryant
+          ID: 0a22bab3de2d83d723811e3fb1ebca904e924a97
+    FQDN: login-test.osgconnect.net
+    DN: /DC=org/DC=incommon/C=US/ST=IL/L=Chicago/O=University of Chicago/OU=IT Services - Self Enrollment/CN=condor-flocking-cert.grid.uchicago.edu
+    Services:
+      Submit Node:
+        Description: OS Pool access point
```
Comment (matyasselmeci) [inline topology/University of Chicago/UChicago/UChicago_OSGConnect_ITB.yaml:29]:
```suggestion
        Description: ITB OS Pool access point
```
maybe? They are different pools after all.

File: topology/University of Wisconsin Milwaukee/UWM - NEMO/NEMO.yaml [hunk 5]
```diff
@@ -54,3 +54,21 @@ Resources:
...
+    ContactLists:
+      Administrative Contact:
+        Primary:
+          Name: Shawn Kwang
+          ID: 115b5c0fb1a160df841dde4591b96c180ba6d778
+      Security Contact:
+        Primary:
+          Name: Shawn Kwang
+          ID: 115b5c0fb1a160df841dde4591b96c180ba6d778
+    FQDN: submit-dev.ligo.uwm.edu
```
Comment (matyasselmeci) [inline topology/University of Wisconsin Milwaukee/UWM - NEMO/NEMO.yaml:71]: I don't see this in flocking-hosts.yml. Is that expected?

Comment (brianhlin) [reply topology/University of Wisconsin Milwaukee/UWM - NEMO/NEMO.yaml]: Good catch, fixed

**Comment by matyasselmeci** [comment]
The half I checked looks good.

File: topology/Rutgers University/Rutgers OARC/SITE.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+AddressLine1: 136 Frelinghuysen Road
+City: Piscataway
+Country: United States
+ID: 10307
+Latitude: 40.522918
+Longitude: -74.464514
+State: New Jersey
+Zipcode: 08854-8019
```
Comment (matyasselmeci) [inline topology/Rutgers University/Rutgers OARC/SITE.yaml:8]:
```suggestion
Zipcode: '08854-8019'
```
probably not necessary in this one case, but it's a good habit to keep up.

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml [hunk 7]
```diff
@@ -43,6 +44,7 @@ Resources:
           Name: Kurt Strosahl
     Description: Submit node for JLab
     FQDN: scosg20.jlab.org
+    DN: /CN=scosg16.jlab.org
```
Comment (matyasselmeci) [inline topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml:47]:
```suggestion
    DN: /CN=scosg20.jlab.org
```

Comment (matyasselmeci) [reply topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml:47]: Also flocking-hosts.yml has an InCommon cert for this resource.

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml [hunk 8]
```diff
@@ -43,6 +44,7 @@ Resources:
...
     ID: 1134
```
Comment (matyasselmeci) [inline topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml:48]: I realize this ID is not part of the change, but 1134 is already used by a Lamar-Cluster resource.

File: topology/University of California San Diego/San Diego Supercomputer Center/SDSC-NSG.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+---
+Production: true
+SupportCenter: Self Supported
+GroupDescription: Resources related at SDSC for the Neuroscience Gateway
+GroupID: 1130
+Resources:
+  SDSC-NSG-AP1:
+    Active: true
+    Description: The Neuroscience gateway access point for the OS Pool
+    ID: 1131
```
Comment (matyasselmeci) [inline topology/University of California San Diego/San Diego Supercomputer Center/SDSC-NSG.yaml:10]: 1131 is already used by a BU_ATLAS_Tier2 resource.

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml (hunk 8 above)
Comment (matyasselmeci) [reply topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml]: We test for name uniqueness and ID presence, but not ID uniqueness. I wonder why...

File: topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml (hunk 2 above)
Comment (brianhlin) [reply topology/University of Chicago/UChicago/UChicago_OSGConnect.yaml:16]: Hrm, I thought I had left a comment: @rynge will be working on ripping out `flocking-hosts.yaml` in favor of what's in Topology so I don't think it's necessary.

**Comment by matyasselmeci** [comment]
I realized that the split was unfair so I checked Purdue through UCSD as well. Aside from my comments, they looked good.

**Review by edquist — COMMENTED** [review]
See comments.

Also, i am just reviewing you changes.  I have not tried to do any deep analysis of flocking-hosts to try to see if you are missing any resources.

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml (hunk 8 above)
Comment (edquist) [reply topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml]: Oh huh.  Yeah i suppose we should add a unique ID check for the different types of things, ey?

File: topology/Harvard Medical School/HMS/HMS_SBGrid.yaml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+          ID: d5ebc623588d1696614076faf0257d83806891a2
+      Security Contact:
+        Primary:
+          Name: SBGrid HPC Mailing List
+          ID: 1777384243c590f66f0ef6b11c93477c80e3062d
+        Secondary:
+          Name: Peter Meyer
+          ID: d5ebc623588d1696614076faf0257d83806891a2
+    FQDN: glidein-1.sbgrid.org
+    DN: /CN=glidein-1.sbgrid.org
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply topology/Harvard Medical School/HMS/HMS_SBGrid.yaml:26]: Let's Encrypt DNs only have a CN.

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml (hunk 8 above)
Comment (brianhlin) [reply topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml]: I don't think we can assume that: the registered FQDN may correspond to one of the cert's SANs.

File: topology/Harvard Medical School/HMS/HMS_SBGrid.yaml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+          Name: Peter Meyer
+          ID: d5ebc623588d1696614076faf0257d83806891a2
+      Security Contact:
+        Primary:
+          Name: SBGrid HPC Mailing List
+          ID: 1777384243c590f66f0ef6b11c93477c80e3062d
+        Secondary:
+          Name: Peter Meyer
+          ID: d5ebc623588d1696614076faf0257d83806891a2
+    FQDN: glidein-1.sbgrid.org
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply topology/Harvard Medical School/HMS/HMS_SBGrid.yaml:25]: Yeah, it's an issue with "site" as an overloaded term. SBGrid is a consortium so feasibly there could be many other institutions with SBGrid resources so I thought that its own resource group in the HMS site would be most reasonable.

File: topology/American Museum of Natural History/AMNH/AMNH.yaml [hunk 12]
```diff
@@ -253,3 +253,29 @@ Resources:
...
+          ID: 0da9ca0b7d1d58f1a91c6797e4bb29b64846f88e
+        Secondary:
+          Name: AMNH OSG Contact List
+          ID: e7b8d36d684570dcb0ed8ff7b723928d5a93b513
+    FQDN: huxley-osgsub-001.sdmz.amnh.org
+    Services:
+      Submit Node:
+        Description: OS Pool access point
+    Tags:
+      - CC*
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply topology/American Museum of Natural History/AMNH/AMNH.yaml:281]: Yup, this should be associated with the CC* stuff at AMNH

File: topology/Illinois Institute of Technology/IIT - Illinois Institute of Technology/IIT.yaml [hunk 13]
```diff
@@ -72,6 +72,7 @@ Resources:
           Name: Grigoriy Abramov
     Description: Illinois Institute of Technology
     FQDN: otsgrid.iit.edu
+    DN: /CN=otsgrid.iit.edu
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply topology/Illinois Institute of Technology/IIT - Illinois Institute of Technology/IIT.yaml:75]: Another GlideinWMS site vs Topology site difference

File: topology/Massachusetts Institute of Technology/MIT_SUBMIT/MIT_SUBMIT.yaml [hunk 14]
```diff
@@ -35,6 +35,7 @@ Resources:
           Name: Max Goncharov
     Description: This is the submission node to MIT on-campus, OSG, and CMS resources.
     FQDN: submit.mit.edu
+    DN: /DC=org/DC=incommon/C=US/ST=MA/L=Cambridge/O=Massachusetts Institute of Technology/OU=Information Systems & Technology/CN=submit.mit.edu
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply topology/Massachusetts Institute of Technology/MIT_SUBMIT/MIT_SUBMIT.yaml:18]: Good catch on the wrong contact. I think the other things you brought up are fine as they are in the PR

File: topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml (hunk 8 above)
Comment (edquist) [reply topology/Thomas Jefferson National Accelerator Facility/JLAB/JLab-Farm.yaml]: Ah ok.

Also:

>It's concerning that our CI doesn't catch this :(

>We test for name uniqueness and ID presence, but not ID uniqueness. I wonder why...

Apparently there is a ticket for this - SOFTWARE-3287 specifies a few new unique ID requirements

**Review by edquist — APPROVED** [review]
No further objections for my half

**Comment by brianhlin** [comment]
Rebased off `master` and force-pushed
